From 6701a248d2c578a85d9a301d3a63b4257e28f4f7 Mon Sep 17 00:00:00 2001
From: Linu Cherian <lcherian@marvell.com>
Date: Tue, 5 May 2020 17:36:15 +0530
Subject: [PATCH 27/37] coresight: etr: Fix CSETR_QUIRK_BUFFSIZE_8BXA quirk
 check

commit 83eeab7daa904adae47c8204c4358e3cc8b320da from
git@git.assembla.com:cavium/WindRiver.linux.git

Fix CSETR_QUIRK_BUFFSIZE_8BXA quirk check.

Change-Id: I9fdc90e138e5138188b203582838ee2064d3371d
Signed-off-by: Linu Cherian <lcherian@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/27915
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[Kevin: Just some minor context mods in order to port to linux-yocto]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/hwtracing/coresight/coresight-tmc-etr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/hwtracing/coresight/coresight-tmc-etr.c b/drivers/hwtracing/coresight/coresight-tmc-etr.c
index d234cb7a753a..361b4b6ef389 100644
--- a/drivers/hwtracing/coresight/coresight-tmc-etr.c
+++ b/drivers/hwtracing/coresight/coresight-tmc-etr.c
@@ -1206,7 +1206,7 @@ static void __tmc_etr_enable_hw(struct tmc_drvdata *drvdata)
 	tmc_wait_for_tmcready(drvdata);
 
 	writel_relaxed(etr_buf->size / 4, drvdata->base + TMC_RSZ);
-	if (drvdata && CSETR_QUIRK_BUFFSIZE_8BX)
+	if (drvdata->etr_options & CSETR_QUIRK_BUFFSIZE_8BX)
 		writel_relaxed(etr_buf->size / 8, drvdata->base + TMC_RSZ);
 	else
 		writel_relaxed(etr_buf->size / 4, drvdata->base + TMC_RSZ);
-- 
2.17.1

